instrumentation/runtime-telemetry/runtime-telemetry-java8/javaagent/src/main/java/io/opentelemetry/instrumentation/javaagent/runtimemetrics/java8/JarAnalyzerUtil.java

```diff
@@ -0,0 +1,180 @@
+   */
+  static void addPackageType(AttributesBuilder builder, URL archiveUrl) throws Exception {
+    String path = archiveUrl.getFile();
+    int extensionStart = path.lastIndexOf(".");
+    if (extensionStart > -1) {
+      builder.put(PACKAGE_TYPE, path.substring(extensionStart + 1));
+      return;
+    }
+    throw new Exception("Cannot extract archive type from URL: " + archiveUrl);
+  }
+
+  /**
+   * Set the attributes {@link #PACKAGE_CHECKSUM} from the {@code archiveUrl}.
+   *
+   * <p>The {@link #PACKAGE_CHECKSUM} is set to the SHA-1 checksum of the archive, e.g. {@code
+   * 30d16ec2aef6d8094c5e2dce1d95034ca8b6cb42}.
+   */
+  static void addPackageChecksum(AttributesBuilder builder, URL archiveUrl) throws IOException {
+    builder.put(PACKAGE_CHECKSUM, computeSha1(archiveUrl));
+  }
+
+  private static String computeSha1(URL archiveUrl) throws IOException {
+    MessageDigest md = MESSAGE_DIGEST_THREAD_LOCAL.get();
+    md.reset(); // Reset reused thread local message digest instead
+
+    try (InputStream is = new DigestInputStream(archiveUrl.openStream(), md)) {
+      byte[] buffer = new byte[1024 * 8];
+      // read in the stream in chunks while updating the digest
+      while (is.read(buffer) != -1) {}
+
+      byte[] mdbytes = md.digest();
+
+      // convert to hex format
+      StringBuilder sb = new StringBuilder(40);
+      for (byte mdbyte : mdbytes) {
+        sb.append(Integer.toString((mdbyte & 0xff) + 0x100, 16).substring(1));
+      }
+
+      return sb.toString();
+    }
+  }
+
+  /**
+   * Set the attributes {@link #PACKAGE_PATH} from the {@code archiveUrl}.
+   *
+   * <p>The {@link #PACKAGE_PATH} is set to the archive file name, e.g. {@code
+   * jackson-datatype-jsr310-2.15.2.jar}.
+   */
+  static void addPackagePath(AttributesBuilder builder, URL archiveUrl) throws Exception {
+    builder.put(PACKAGE_PATH, archiveFilename(archiveUrl));
+  }
+
+  private static String archiveFilename(URL archiveUrl) throws Exception {
+    String path = archiveUrl.getFile();
+    int start = path.lastIndexOf(File.separator);
+    if (start > -1) {
+      return path.substring(start + 1);
+    }
+    throw new Exception("Cannot extract archive file name from archive URL: " + archiveUrl);
+  }
+
+  /**
+   * Set the attributes {@link #PACKAGE_DESCRIPTION} from the {@code archiveUrl}.
+   *
+   * <p>The {@link #PACKAGE_DESCRIPTION} is set to manifest "{Implementation-Title} by
+   * {Implementation-Vendor}", e.g. {@code Jackson datatype: JSR310 by FasterXML}.
+   */
+  static void addPackageDescription(AttributesBuilder builder, URL archiveUrl) throws IOException {
+    try (JarFile jarFile = new JarFile(archiveUrl.getFile())) {
+      Manifest manifest = jarFile.getManifest();
+      if (manifest == null) {
+        return;
+      }
+
+      java.util.jar.Attributes mainAttributes = manifest.getMainAttributes();
+      String name = mainAttributes.getValue(java.util.jar.Attributes.Name.IMPLEMENTATION_TITLE);
```

Comment:
I noticed you're not getting `Implementation-Version`.  Is that intentional?

Reply:
The version is being grabbed from the pom, so yes intentional. Although I don't feel particularly strongly about how the `package.description` field is derived.